src/gui/qgsnewhttpconnection.cpp

```diff
@@ -330,3 +331,8 @@ void QgsNewHttpConnection::accept()
 
   QDialog::accept();
 }
+
+void QgsNewHttpConnection::helpRequest()
+{
+  QgsHelp::openHelp( "managing_data_source/opening_data.html#connecting-to-web-services" );
```

Comment:
Actually, this section is just to list in that page full types of data opened by QGIS. It acts like a shortcut so better use the section that treats the subject and avoid one more click to readers. The most relevant page for this dialog might be somewhere in http://docs.qgis.org/testing/en/docs/user_manual/working_with_ogc/index.html